### DISCUSSION TITLE
Command level metrics

### DISCUSSION THREAD

**Discussion by Harleen13Singh** [redis/rueidis#705] [Q&A]
Do we have any way of getting the command level metrics? Latency and qps? 
I can see in the metrics.go file, only dial level metrics are exposed.
One way is to wrap methods and publish the metrics by myself, but wondering if we are exposing somewhere?
Also AWS does not provides command level logs

**Comment by proost** [comment]
is [redis_exporter](https://github.com/oliver006/redis_exporter) not enough?

**Reply by Harleen13Singh** [reply]
Thanks for recommending this!!

**Comment by rueian** [comment]
We don't have command-level metrics because that usually results in too many labels for metric systems, but we have command-level traces in `rueidisotel`. You can turn your traces into metrics with some post-processing.

**Reply by Harleen13Singh** [reply]
Understood. Thanks!
